File: app/src/main/java/com/imangazalievm/bubbble/di/modules/ShotDetailsPresenterModule.java
package com.imangazalievm.bubbble.di.modules;

import com.imangazalievm.bubbble.di.scopes.Presenter;

import dagger.Module;
import dagger.Provides;

@Module
public class ShotDetailsPresenterModule {

    private final long shotId;

    public ShotDetailsPresenterModule(long shotId) {
        this.shotId = shotId;
    }

    @Provides
    @Presenter
    long provideShotId() {
        return shotId;
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/di/shotdetails/ShotDetailsPresenterModule.kt
package com.imangazalievm.bubbble.di.shotdetails

import com.imangazalievm.bubbble.di.global.scopes.Presenter
import dagger.Module
import dagger.Provides

@Module
class ShotDetailsPresenterModule(private val shotId: Long) {

    @Provides
    @Presenter
    fun provideShotId(): Long = shotId

}